django_deprecate_fields/deprecate_field.py

```diff
@@ -11,28 +13,28 @@ class DeprecatedField(object):
     deprecate_field function instead.
     """
 
-    def __init__(self, val):
+    def __init__(self, val=None, use_current=False, field_name=None):
+        self.use_current = use_current
+        self.field_name = field_name
         self.val = val
 
-    def _get_name(self, obj):
-        """
-        Try to find this field's name in the model class
-        """
-        for k, v in type(obj).__dict__.items():
-            if v is self:
-                return k
-        return "<unknown>"
-
-    def __get__(self, obj, type=None):
+    def __get__(self, instance, type=None):
         msg = "accessing deprecated field %s.%s" % (
-            obj.__class__.__name__,
-            self._get_name(obj),
+            instance.__class__.__name__,
+            self.field_name,
```

Comment:
@jackton1 Could you state why you can exchange `_get_name` with `field_name` here? To be honest, I don't remember the purpose of the logic in `_get_name`. But did you figure it out?